firehose.go

```diff
@@ -0,0 +1,94 @@
+package hedwig
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+type Firehose struct {
+	messageValidator *messageValidator
+}
+
+func (f *Firehose) Deserialize(reader io.Reader) ([]Message, error) {
+	runWithTransportMessageAttributes := false
+	var messages []Message
+	if f.messageValidator.encoder.IsBinary() {
+		var messagePayload []byte
+		for {
+			// TLV format: 8 bytes for size of message, n bytes for the actual message
+			msgSize := make([]byte, 8)
+			l, err := reader.Read(msgSize)
+			if err == io.EOF && l != 0 {
+				return nil, fmt.Errorf("TLV may be malformed EOF reached when trying to read length")
+			} else if err == io.EOF {
+				break
+			} else if err != nil {
+				return nil, err
+			}
+			msgLength := binary.LittleEndian.Uint64(msgSize)
+			if len(messagePayload) < int(msgLength) {
+				messagePayload = append(messagePayload, make([]byte, int(msgLength)-len(messagePayload))...)
+			}
+			_, err = reader.Read(messagePayload)
```

Comment:
```suggestion
			_, err = reader.Read(messagePayload[:msgLength])
```
since `msgLength` could be less than `len(messagePayload)`